Copy active inputs between CAs if they have the same set of inputs

Arnei opened the pull request:
For a scheduled event in the admin ui, the capture agent for the event can be changed in the event details. However, when changing the capture agent, the input channels for the capture agent will always be unset (except for the capture agent the event was created with). As most institutions usually use multiple capture agents of the same brand, with the same set of input channels, having to manually reactivate every input after switching capture agents is quite cumbersome.

This PR changes the default behaviour for switching capture agents. If both capture agents have the same set of input channels, which inpus are active will now be copied over from the previously selected capture agent to the newly selected one.

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

Arnei committed:
Copy active inputs between CAs if they have the same set of inputs

For a scheduled event in the admin ui, the capture agent for the event
can be changed in the event details. However, when changing the capture
agent, the input channels for the capture agent will always be unset
(except for the capture agent the event was created with). As most
institutions usually use multiple capture agents of the same brand,
with the same set of input channels, having to manually reactivate
every input after switching capture agents is quite cumbersome.

This PR changes the default behaviour for switching capture agents.
If both capture agents have the same set of input channels, which
inpus are active will now be copied over from the previously selected
capture agent to the newly selected one.

snoesberger commented:
The change basically works, the SMP input channel settings are now copied over. However, there is still one flaw: at the first moment, after changing the CA, the two checkboxes are deactivated. After the change has been processed (takes in our environment a few seconds), the checkboxes are active again (as they were before).

Is it possible to adjust this behaviour so that the checkboxes remain activated after the CA change until processing is complete?

Arnei committed:
Fix nullpointer exception

oldObj is not necessarily defined

Arnei committed:
Prefill ca input channels if appropriate

When switching between CAs with the exact same input channels, the
currently active input channels should be copied over to the new CA.
However, for large installations the response from the backend may take
a couple seconds, until which the input channels show as deactivated.
This should fix that.

Arnei committed:
Fix eslint complaints

Arnei commented:
@snoesberger Thanks for the feedback, should now behave like you want it to.

snoesberger commented:
@Arnei Thanks for the quick fix!
I just found another issue. On an existing event with an Extron SMP as CA, I can't change the CA to a pyca. I assume this is because pyca's do not have channel attributes like the Extron SMP's do?
In the broser console I can see that the following request leads to a 500 Internal Server Error:
https://admin.opencast-test.unibe.ch/admin-ng/event/65f3a50a-bf2f-4d35-8ae4-10bd47fe257d/scheduling
With data:
```
{
  "agentId":"pyca-Sascha",
  "start":"2023-08-31T15:00:00Z",
  "end":"2023-08-31T15:10:00Z",
  "agentConfiguration":
  {
    "event.series":"c23a27cc-8eb9-41e7-8965-d01d76b81794",
    "capture.device.names":"",
    "org.opencastproject.workflow.config.publishLive":"true",
    "org.opencastproject.workflow.definition":"schedule-and-upload",
    "org.opencastproject.workflow.config.straightToPublishing":"false",
    "event.location":"WTI-Hal6-020-ext351L",
    "event.title":"Live Stream 2023-08-31 17:00"
  },
  "previousAgentId":"WTI-Hal6-020-ext351L",
  "previousEntries":
  {
    "Channel A":true,"Channel B":true
  }
}
```

In the opencast server logs I have the following warning at the same time:
`2023-08-31T16:23:05,170 | WARN  | (LogUtils:468) - Application {http://endpoint.external.opencastproject.org/}ListProviderEndpoint has thrown exception, unwinding now
org.apache.cxf.interceptor.Fault: null must not be wrapped in a some`

Arnei committed:
Fix null wrapping exception

Exception occured when passing null to `Opt.some()`. Fixed by not using
entwine.

Arnei commented:
Thanks for the detailed report, should be fixed now.
